go/runtime/txpool/local_queue.go

```diff
@@ -0,0 +1,80 @@
+package txpool
+
+import (
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+)
+
+var (
+	_ UsableTransactionSource        = (*localQueue)(nil)
+	_ RecheckableTransactionStore    = (*localQueue)(nil)
+	_ RepublishableTransactionSource = (*localQueue)(nil)
+)
+
+// localQueue is a "front of the line" area for txs from our own node. We also keep these txs in order.
+type localQueue struct {
```

Comment:
Probably needs synchronization?

Reply:
I need to examine what goroutines interact with it

Reply:
added a lock